packages/util-endpoints/src/resolveEndpoint.ts

```diff
@@ -31,5 +31,19 @@ export const resolveEndpoint = (ruleSetObject: RuleSetObject, options: EndpointR
     }
   }
 
-  return evaluateRules(rules, { endpointParams, logger, referenceRecord: {} });
+  const endpoint = evaluateRules(rules, { endpointParams, logger, referenceRecord: {} });
+
+  if (options.endpointParams?.Endpoint) {
+    // take protocol and port from custom Endpoint if present.
+    try {
+      const givenEndpoint = new URL(options.endpointParams.Endpoint as string);
+      const { protocol, port } = givenEndpoint;
+      endpoint.url.protocol = protocol;
+      endpoint.url.port = port;
+    } catch (e) {
+      // ignored
+    }
+  }
```

Comment:
this could simply be a problem with the sample ruleset (not shown in this PR)

the resolved endpoint template starts with a hard-coded `https://{...` but a test case expects the protocol of the custom `endpointParams.Endpoint` to be respected.